src/Compiler/TypedTree/TypedTreeOps.fs

```diff
@@ -3513,12 +3513,25 @@ let HasILAttribute tref (attrs: ILAttributes) =
 let TryDecodeILAttribute tref (attrs: ILAttributes) = 
     attrs.AsArray() |> Array.tryPick (fun x -> if isILAttrib tref x then Some(decodeILAttribData x) else None)
 
+let TryDecodeILAttributeByName nm (attrs: ILAttributes) = 
+    attrs.AsArray() |> Array.tryPick (fun x -> if isILAttribByName ([], nm) x then Some(decodeILAttribData x) else None)
+
 // F# view of attributes (these get converted to AbsIL attributes in ilxgen) 
 let IsMatchingFSharpAttribute g (AttribInfo(_, tcref)) (Attrib(tcref2, _, _, _, _, _, _)) = tyconRefEq g tcref tcref2
 let HasFSharpAttribute g tref attrs = List.exists (IsMatchingFSharpAttribute g tref) attrs
 let TryFindFSharpAttribute g tref attrs = List.tryFind (IsMatchingFSharpAttribute g tref) attrs
 let TryFindFSharpAttributeOpt g tref attrs = match tref with None -> None | Some tref -> List.tryFind (IsMatchingFSharpAttribute g tref) attrs
 
+let TryFindFSharpAttributeByName nm attrs = 
```

Comment:
Consider: Existing mechanism via `TryFindFSharpAttribute` and having a reference to the attribute in `TcGlobals`, i.e. work on strongly typed referenced to attribute definitions and not on strings.